apps/src/templates/instructions/codeReview/CommentEditor.jsx

```diff
@@ -1,19 +1,35 @@
 import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import $ from 'jquery';
+import {getStore} from '@cdo/apps/redux';
 import msg from '@cdo/locale';
 import javalabMsg from '@cdo/javalab/locale';
 import Button from '@cdo/apps/templates/Button';
 import color from '@cdo/apps/util/color';
 
 export default class CommentEditor extends Component {
-  state = {
-    comment: '',
-    isEditing: false
+  static propTypes = {
+    onNewCommentSubmit: PropTypes.func.isRequired,
+    token: PropTypes.string.isRequired
   };
 
-  commentChanged = event => {
-    this.setState({
-      comment: event.target.value,
-      isEditing: true
+  state = {comment: ''};
+
+  commentChanged = event => this.setState({comment: event.target.value});
+
+  submitNewComment = () => {
+    $.ajax({
```

Comment:
Move the ajax call to ReviewTab or TopInstructions. 

Reasoning: CommentEditor is set up very similarly to a [controlled component](https://reactjs.org/docs/forms.html). This makes a lot of sense and we should lean into this and go ahead and make it a controlled component. Since it's set up as a controlled component, however, I was a surprised to see the ajax call here. I kept looking for it in TopInstructions (where the other ajax call is) or in ReviewTab (which controls the state of the comment list) and had assumed it hadn't been implemented yet. Moving the ajax call to one of those other locations will make the behavior a bit more expected for future maintainers.